util/topgen.py

```diff
@@ -990,6 +1009,11 @@ def main():
     else:
         outdir = Path(args.outdir)
 
+    alt_hjson_path = ""
+    if args.hjson_path:
+        alt_hjson_path = args.hjson_path
+        log.info("Alternate hjson path is {alt_hjson_path}")
```

Comment:
I'd suggest getting rid of the `alt_hjson_path` variable and just passings `args.hjson_path` below. Then this code could be:
```suggestion
    if args.hjson_path is not None:
        log.info("Alternate hjson path is {args.hjson_path}")
```
That way, we get more helpful behaviour if someone messes up and types `--hjson-path=`.

Finally, change the checks above from `if reggen_only and alt_hjson_path` to `if reggen_only and alt_hjson_path is not None`.

Reply:
i know you mentioned this before, but can you remind me again the benefits of using an explicit `is not None` check vs just using the variable directly? Is it to make sure an accidental change in type don't just break things later?

Reply:
get rid of the variable and used `is not None`. I just realized you also explained that this will help against the accidentally inputting empty string case.